meerk40t/extra/vectrace.py

```diff
@@ -144,3 +123,234 @@ def _vectrace(pixels, width, height):
+    RANDOM:     choose randomly.
+    The current default policy is MINORITY, which tends to keep
+    visual lines connected.
+
+-   The alphamax parameter is a threshold for the detection of corners.
+    It controls the smoothness of the traced curve.
+    The current default is 1.0; useful range of this parameter is
+    from 0.0 (polygon) to 1.3333 (no corners).
+
+-   The opticurve parameter is a boolean flag that controls whether Potrace
+    will attempt to “simplify” the final curve by reducing the number of
+    Bezier curve segments. Opticurve=1 turns on optimization,
+    and opticurve=0 turns it off. The current default is on.
+
+-   The opttolerance parameter defines the amount of error
+    allowed in this simplification. The current default is 0.2. Larger values tend to decrease the number of segments, at the expense of less accuracy. The useful range is from 0 to infinity, although in practice one would hardly choose values greater than 1 or so. For most purposes, the default value is a good tradeoff between space and accuracy.
+
+"""
+
+def setup_potrace(kernel):
+    _ = kernel.translation
+    import potrace
+    import numpy
+
+    @kernel.console_option(
+        "turnpolicy",
+        "z",
+        type=str,
+        default="minority",
+        help=_("how to resolve ambiguities in path decomposition"),
+    )
+    @kernel.console_option(
+        "turdsize",
+        "t",
+        type=int,
+        default=2,
+        help=_("suppress speckles of up to this size (default 2)"),
+    )
+    @kernel.console_option(
+        "alphamax",
+        "a",
+        type=float,
+        default=1,
+        help=_("corner threshold parameter")
+    )
+    @kernel.console_option(
+        "opticurve",
+        "n",
+        type=bool,
+        action="store_true",
+        help=_("turn off curve optimization"),
+    )
+    @kernel.console_option(
+        "opttolerance",
+        "O",
+        type=float,
+        help=_("curve optimization tolerance"),
+        default=0.2,
+    )
+    @kernel.console_option(
+        "color",
+        "C",
+        type=Color,
+        help=_("set foreground color (default Black)"),
+    )
+    @kernel.console_option(
+        "invert",
+        "i",
+        type=bool,
+        action="store_true",
+        help=_("invert bitmap"),
+    )
+    @kernel.console_option(
+        "blacklevel",
+        "k",
+        type=float,
+        default=0.5,
+        help=_("blacklevel?!"),
+    )
+    @kernel.console_command(
+        "potrace",
+        help=_("return paths around image"),
+        input_type="image",
+        output_type="elements",
+    )
+    def do_potrace(
+        data,
+        turnpolicy=None,
+        turdsize=None,
+        alphamax=None,
+        opticurve=None,
+        opttolerance=None,
+        color=None,
+        invert=None,
+        blacklevel=None,
+        **kwargs,
+    ):
+        if turnpolicy not in (
```

Comment:
While not wrong, this should probably be a function that gets registered into the lookup. Or something that would let us access the code through a text-to-path routine.